### PR TITLE
Swap gosdk functions

### PR DESCRIPTION
**Pull request by moldis**
### Changes
-

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: zcnswap/config/config.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+var Configuration SwapConfig
+
+type SwapConfig struct {
+	BancorAddress    string
+	UsdcTokenAddress string
+	ZcnTokenAddress  string
+	WalletMnemonic   string
+}
+
+func Init(file string) error {
```
Comment (dabasov) [inline zcnswap/config/config.go:20]: will it work on mobile?

File: zcnswap/swapfactory/bancor/service.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	value := big.NewInt(0)
+	opts, err := ethutils.NewSignedTransaction(convertPayload, from, spender.Hex(), value, s.PrivateKey, s.Client)
+	if err != nil {
+		l.Logger.Error("Signed transaction errorr", zap.Error(err))
+		return nil, err
+	}
+
+	// ropsten test network deployed old contract, hence using old method
+	//return bancorModule.ConvertByPath2(opts, *path, amountIn, amountOut, affiliateAccount)
+	return bancorModule.ConvertByPath(opts, *pair.ConversionPath, pair.AmountIn, pair.AmountOut, affiliateAccount, affiliateAccount, big.NewInt(0))
```
Comment (dabasov) [inline zcnswap/swapfactory/bancor/service.go:93]: move to config so we could change network for mainnet

**Review by dabasov — CHANGES_REQUESTED** [review]

File: zcnswap/config/config.go (hunk 1 above)
Comment (moldis) [reply zcnswap/config/config.go:20]: Yes

File: zcnswap/swapfactory/bancor/service.go (hunk 2 above)
Comment (moldis) [reply zcnswap/swapfactory/bancor/service.go:93]: It depends on **eth client** network, which is initialized upper in the code.

`func Swap(swapAmount int64, tokenSource string) (string, error) {
	client, _ := zcncore.GetEthClient()`

https://github.com/0chain/gosdk/blob/91bf935b5282d4deb7882c05eaafb6ab48e719d4/zcncore/ethwallet.go#L30

This code already pickup network from config

Comment (dabasov) [reply zcnswap/swapfactory/bancor/service.go:93]: remove commented code then

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/zcnswap  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]

**Pull request merged**